src/wasm-interpreter.h

```diff
@@ -75,11 +75,45 @@ typedef std::vector<Literal> LiteralList;
 
 // Debugging helpers
 #ifdef WASM_INTERPRETER_DEBUG
-#define NOTE_ENTER(x) { std::cout << "visit " << x << " : " << curr << "\n"; }
-#define NOTE_ENTER_(x) { std::cout << "visit " << x << "\n"; }
-#define NOTE_NAME(p0) { std::cout << "name " << '('  << Name(p0) << ")\n"; }
-#define NOTE_EVAL1(p0) { std::cout << "eval " << '('  << p0 << ")\n"; }
-#define NOTE_EVAL2(p0, p1) { std::cout << "eval " << '('  << p0 << ", " << p1 << ")\n"; }
+class Indenter {
+static int indentLevel;
+
+const char* _str;
+
+public:
+  Indenter(const char* str) : _str(str) {
+    ++indentLevel;
+  }
+  ~Indenter() {
+    print();
+    std::cout << "exit " << _str << '\n';
+    --indentLevel;
+  }
+
+  static void print() {
+    std::cout << indentLevel << ':';
+    for (int i = 0; i <= indentLevel; ++i) {
+      std::cout << ' ';
+    }
+  }
+};
+int Indenter::indentLevel = 0;
```

Comment:
this looks worrying, if the include is included in multiple files it will conflict at link time

do we actually need a static variable here? if we do, we could define it in a single cpp file (i guess `wasm.cpp` for now, although it would be nice to have `wasm-interpreter.cpp` eventually)